packages/core/src/modules/connections/DidExchangeProtocol.ts

```diff
@@ -164,7 +166,7 @@ export class DidExchangeProtocol {
       )
     }
 
-    const didDocument = await this.extractDidDocument(message)
+    const didDocument = await this.extractDidDocument(messageContext.agentContext, message)
```

Comment:
Why exactly is the agentContext inside the messageContext? Just curious.

Reply:
Because each message can have their own context. Handlers, services and repositories are now all stateless. The agent context gives the context to process the message. So if we receive a message we'll find the correct tenant, assign the agent context for that tenant to the message context and then process the message. 

We could make it a separate parameter, but it kinda made sense to me to tie the agent context to a specific message context. Would you prefer to make it a separate parameter?

Reply:
Its a difficult one. I think for consistency a seperate `agentContext` might be nicer, but I see your point of containing the agentContext inside the messageContext. I am okay with keeping it like this.